### PR TITLE
ToS cut off at GD

### PR DESCRIPTION
**Pull request by swomma**
Solves 29th/personnel#251

### CODE REVIEW

File: application/controllers/assignments.php [hunk 1]
```diff
@@ -29,18 +30,24 @@ public function index_get($member_id = FALSE) {
...
+    private function calculate_duration($assignments,$member_id) {
         $days = array();
+        $this->discharge_model->where('type','General');
+        $this->discharge_model->where('discharges.member_id',$member_id);
+        $gdDate = $this->discharge_model->get()->result_array();
         foreach($assignments as $assignment) {
             $start_date = strtotime($assignment['start_date']);
             $end_date = strtotime($assignment['end_date'] ?: format_date('now', 'mysqldate'));
             for($i = $start_date; $i < $end_date; $i = $i + DAY) {
+              if ( format_date($i, 'mysqldate') > $gdDate[0]['date'] )
```
Comment (wilson29thid) [inline application/controllers/assignments.php]: Having this inside the `for` loop on L#46 makes it format the date and compare for every **day** between the `start_date` and `end_date`. Alternatively, we could just check if `start_date > $gdDate[0]['date']` prior to the loop.

File: application/controllers/assignments.php [hunk 2]
```diff
@@ -29,25 +30,32 @@ public function index_get($member_id = FALSE) {
...
-            $this->response(array('status' => true, 'duration' => $duration, 'assignments' => $assignments));
+            $duration = $this->calculate_duration($assignments,$member_id);
+            $this->response(array('status' => true, 'duration' => $duration[0], 'gddate' => $duration[1], 'assignments' => $assignments ));
         }
     }
     
-    private function calculate_duration($assignments) {
+    private function calculate_duration($assignments,$member_id) {
         $days = array();
+        $this->discharge_model->where('type','General');
```
Comment (wilson29thid) [inline application/controllers/assignments.php]: This should probably be where type is not Honorable

**Comment by wilson29thid** [comment]
06:04 - Lt. Col. Wilson [29th ID]: also, just thought of something
06:04 - Lt. Col. Wilson [29th ID]: if duration already takes into account gddate, do we even need to include gddate in the API response?
06:04 - Lt. Col. Wilson [29th ID]: can't the client (app) simply calculate date - duration (days) ?

**Comment by swomma** [comment]
Changed to !=Honrable

**Pull request merged**